src/jit/lsra.cpp

```diff
@@ -7551,6 +7660,15 @@ void LinearScan::allocateRegisters()
             currentInterval->physReg               = assignedRegister;
             regsToFree &= ~assignedRegBit; // we'll set it again later if it's dead
 
+#ifdef _TARGET_ARM_
+            // Update overlapping floating point register for TYP_DOUBLE
+            if (currentInterval->registerType == TYP_DOUBLE)
+            {
+                assert(isFloatRegType(currentInterval->assignedReg->registerType));
+                regsToFree &= ~genRegMask(REG_NEXT(assignedRegister));
+            }
+#endif
+
             // If this interval is dead, free the register.
```

Comment:
I've also came to the similar conclusion after implementing `freeRegister()` to handle TYP_DOUBLE correctly, but was not sure.

Thank you for the confirmation :)